tiledb/sm/c_api/tiledb_experimental.h

```diff
@@ -171,6 +174,640 @@ TILEDB_EXPORT int32_t tiledb_array_evolve(
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_free(&file);
+ * @endcode
+ * @param file The file object to destroy
+ */
+TILEDB_EXPORT void tiledb_file_free(tiledb_file_t** file);
+
+/**
+ * Create a file array with default schema
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_t* file;
+ * tiledb_file_alloc(ctx, "s3://tiledb_bucket/my_file", &file);
+ * tiledb_file_create_default(ctx, file);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The tiledb_file_t to be created.
+ * @param config Configuration parameters for the upgrade.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_create_default(
+    tiledb_ctx_t* ctx, tiledb_file_t* file, tiledb_config_t* config);
+
+/**
+ * Create a file array using heuristics based on a file at provided URI
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_t* file;
+ * tiledb_file_alloc(ctx, "s3://tiledb_bucket/my_file", &file);
+ * tiledb_file_create_from_uri(ctx, file, "input_file", NULL);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The file object.
+ * @param input_uri URI to read file from
+ * @param config TileDB Config for setting to create.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_create_from_uri(
+    tiledb_ctx_t* ctx,
+    tiledb_file_t* file,
+    const char* input_uri,
+    tiledb_config_t* config);
+
+/**
+ * Create a file array using heuristics based on a file from provided VFS
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_t* file;
+ * tiledb_file_alloc(ctx, "s3://tiledb_bucket/my_file", &file);
+ *
+ * tiledb_vfs_t vfs*;
+ * tiledb_vfs_alloc(ctx, &vfs);
+ * tiledb_vfs_fh_t *vfs_fh;
+ * tiledb_vfs_open(ctx, vfs, "some_file", TILEDB_VFS_READ, &fh);
+ * tiledb_file_create_from_vfs_fh(ctx, file, vfs_fh, NULL);
+ *
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The file object.
+ * @param input vfs file handle to create from.
+ * @param config TileDB Config for setting to create.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_create_from_vfs_fh(
+    tiledb_ctx_t* ctx,
+    tiledb_file_t* file,
+    tiledb_vfs_fh_t* input,
+    tiledb_config_t* config);
+
+/**
+ * Read a file into the file array from the given FILE handle
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_t* file;
+ * tiledb_file_alloc(ctx, "s3://tiledb_bucket/my_file", &file);
+ * tiledb_file_open(ctx, file, TILEDB_READ);
+ * const char* mime_type;
+ * uint32_t size = 0;
+ * tiledb_file_get_mime_type(ctx, file, *mime_type, &size);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The file object.
+ * @param in FILE handle to read from
+ * @param config TileDB Config for setting to create.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_store_fh(
```

Comment:
Delete. I suggest we have two functions:
* `tiledb_file_import` (to import a file into a "TileDB file")
* `tiledb_file_expert` (to export a "TileDB file" into its original file)

It is a good idea to keep the config as input.

Reply:
Moreover, for `tiledb_file_import` I suggest we maintain the versioning functionality. That is, two invocations of this function should create two fragments in the file. I am sure that this will be useful from the start.